### PR TITLE
Extract method for dynamic mapping update path in TransportShardBulkAction

### PR DESCRIPTION
**Pull request by original-brownbear**
Move this complicated and relatively rarely executed logic to a separate method to enable further refactoring and improve readability.

### CODE REVIEW

**Comment by original-brownbear** [comment]
@elasticmachine update branch

**Comment by original-brownbear** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java [hunk 1]
```diff
@@ -379,64 +378,79 @@ static boolean executeBulkItemRequest(
...
-                    primary.mapperService()
-                        .merge(
-                            MapperService.SINGLE_MAPPING_NAME,
-                            new CompressedXContent(result.getRequiredMappingUpdate()),
-                            MapperService.MergeReason.MAPPING_AUTO_UPDATE_PREFLIGHT
-                        )
-                ).map(DocumentMapper::mappingSource);
-                Optional<CompressedXContent> previousSource = Optional.ofNullable(primary.mapperService().documentMapper())
-                    .map(DocumentMapper::mappingSource);
+        onComplete(result, context, updateResult);
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java:394]: The `onComplete` change: we call it the same way as before, just no longer from an else block. Effectively, this improves readability and is not a functional change, correct?

**Review by javanna — APPROVED** [review]
left a small question to verify my understanding, LGTM though!

File: server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java:394]: Exactly it's the same code just a little drier :)

**Comment by original-brownbear** [comment]
Thanks Luca!

**Pull request merged**